mmtrack/models/trackers/byte_tracker.py

```diff
@@ -0,0 +1,313 @@
+                Defaults to 0.5.
+
+        Returns:
+            tuple(int): The assigning ids.
+        """
+        # get track_bboxes
+        track_bboxes = np.zeros((0, 4))
+        for id in ids:
+            track_bboxes = np.concatenate(
+                (track_bboxes, self.tracks[id].mean[:4][None]), axis=0)
+        track_bboxes = torch.from_numpy(track_bboxes).to(det_bboxes)
+        track_bboxes = self.xyah2xyxy(track_bboxes)
+
+        # compute distance
+        ious = bbox_overlaps(track_bboxes, det_bboxes[:, :4])
+        if weight_iou:
+            ious *= det_bboxes[:, 4][None]
+        dists = (1 - ious).cpu().numpy()
+
+        # bipartite match
+        if dists.size > 0:
+            cost, row, col = lap.lapjv(
+                dists, extend_cost=True, cost_limit=1 - match_iou_thr)
+        else:
+            row = np.zeros(len(ids)).astype(np.int32) - 1
+            col = np.zeros(len(det_bboxes)).astype(np.int32) - 1
+        return row, col
+
+    @force_fp32(apply_to=('img', 'bboxes'))
+    def track(self,
+              img,
+              img_metas,
+              model,
+              bboxes,
+              labels,
+              frame_id,
+              rescale=False,
+              **kwargs):
+        """Tracking forward function.
+
+        Args:
+            img (Tensor): of shape (N, C, H, W) encoding input images.
+                Typically these should be mean centered and std scaled.
+            img_metas (list[dict]): list of image info dict where each dict
+                has: 'img_shape', 'scale_factor', 'flip', and may also contain
+                'filename', 'ori_shape', 'pad_shape', and 'img_norm_cfg'.
+            model (nn.Module): MOT model.
+            bboxes (Tensor): of shape (N, 5).
+            labels (Tensor): of shape (N, ).
+            frame_id (int): The id of current frame, 0-index.
+            rescale (bool, optional): If True, the bounding boxes should be
+                rescaled to fit the original scale of the image. Defaults to
+                False.
+        Returns:
+            tuple: Tracking results.
+        """
+        if not hasattr(self, 'kf'):
+            self.kf = model.motion
+
+        if self.empty or bboxes.size(0) == 0:
+            valid_inds = bboxes[:, -1] > self.init_track_score
+            bboxes = bboxes[valid_inds]
+            labels = labels[valid_inds]
+            num_new_tracks = bboxes.size(0)
+            ids = torch.arange(self.num_tracks,
+                               self.num_tracks + num_new_tracks).to(labels)
+            self.num_tracks += num_new_tracks
+
+        else:
+            # 0. init
+            ids = torch.full((bboxes.size(0), ), -1).to(labels)
+
+            # get the detection bboxes for the first association
+            first_det_inds = bboxes[:, -1] > self.high_det_score
+            (first_det_bboxes, first_det_labels,
+             first_det_ids) = self.get_data_from_inds(
```

Comment:
I misunderstood the usage of this function. Actually we can directly `det_bboxes = det_bboxes[inds]`. If you really want to use some functions, try to use `map()`